ocaml/xapi/xapi_xenops.ml

```diff
@@ -2275,13 +2278,25 @@ let update_vm ~__context id =
               try
                 Option.iter
                   (fun (_, state) ->
-                    debug "xenopsd event: Updating VM %s last_start_time <- %s"
-                      id
-                      (Date.to_string (Date.of_float state.Vm.last_start_time)) ;
                     let metrics = Db.VM.get_metrics ~__context ~self in
                     let start_time = Date.of_float state.Vm.last_start_time in
-                    Db.VM_metrics.set_start_time ~__context ~self:metrics
-                      ~value:start_time ;
+                    if
+                      start_time
+                      <> Db.VM_metrics.get_start_time ~__context ~self:metrics
+                    then (
+                      debug
+                        "xenopsd event: Updating VM %s last_start_time <- %s" id
+                        (Date.to_string (Date.of_float state.Vm.last_start_time)) ;
+                      Db.VM_metrics.set_start_time ~__context ~self:metrics
+                        ~value:start_time ;
+                      if
+                        power_state <> `Suspended (* not VM suspend *)
+                        && power_state_before_update
+                           <> `Suspended (* not VM resume *)
```

Comment:
What about 'force shutdown' on a suspended VM?

Reply:
Thank you for finding this missed scenario, this needs to be added.

Reply:
The function `force_state_reset` is the safest way to clear state (as you did earlier), as it it always called when a VM is shut down or suspended.